src/catchup/IndexBucketsWork.h

```diff
@@ -32,11 +33,13 @@ class IndexBucketsWork : public Work
         bool onAbort() override;
     };
 
+    HistoryArchiveState const& mHas;
+
     bool mWorkSpawned{false};
     void spawnWork();
 
   public:
-    IndexBucketsWork(Application& app);
+    IndexBucketsWork(Application& app, HistoryArchiveState const& has);
```

Comment:
should we pass a vector of bucket pointers here instead? Because indexing can take a while and our policy for bucket garbage collection is kind of arbitrary, I think it's safer to keep strong references to buckets you need

Reply:
Sounds reasonable to me. I'll do a set of pointers instead of a vector though. We already used a set of buckets to avoid duplicate indexing in `IndexBucketsWork`. Having a set as an input parameter means we can get rid of that check.

Reply:
Wait never mind, it's a set of shared pointers, not Buckets, so we still need the check. A vector is better.

Reply:
sounds good, you probably want to have the vector as a member variable of AssumeStateWork though as it can get reset/restarted asynchronously, losing buckets references with the current change.